docs/prerequisites.md

```diff
@@ -0,0 +1,111 @@
+# Pre-Requisites
+
+## CredentialsRequest
+
+### For non-STS clusters
+
+1. Additional credentials are needed for the operator to be successfully
+   installed. This is needed to interact with subnets and VPCs.
+2. Create AWS credentials profile for the operator
+
+    ```bash
+    cat << EOF > credentials
+    [default]
+    aws_access_key_id=${AWS_ACCESS_KEY_ID}
+    aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}
+    EOF
+    
+    oc create secret generic aws-load-balancer-operator \
+    -n aws-load-balancer-operator \
+    --from-file=credentials=credentials
+    ```
+  
+3. Alternatively use the *CredentialsRequest* resource in the `hack` directory
+
+   ```bash
+   oc apply -f https://raw.githubusercontent.com/openshift/aws-load-balancer-operator/main/hack/operator-credentials-request.yaml
+   ```
+
+### Fos STS clusters
+
+The same steps mentioned above can be followed for STS clusters as well.
+But if instead of manually creating the secret in `step 2`, you prefer
+`step 3`, then there are additional steps involved.
+
+1. Post creating the *CredentialRequests*,
+
+    ```bash
+    oc get credentialsrequest -n openshift-cloud-credential-operator  \
+        aws-load-balancer-controller -o yaml > <path-to-credrequests-dir>/cr.yaml
+    ```
+
+    Extract and save the required *CredentialsRequest* in a directory.
+
+2. Use the ccoctl tool to process all *CredentialsRequest* objects in the credrequests
+directory:
+
+    ```bash
+    ccoctl aws create-iam-roles \
+        --name <name> --region=<aws_region> \
+        --credentials-requests-dir=<path-to-credrequests-dir> \
+        --identity-provider-arn <oidc-arn>
+    ```
+
+    For each *CredentialsRequest* object, `ccoctl` creates an IAM role with a trust
+    policy that is tied to the specified OIDC identity provider, and permissions
+    policy as defined in each *CredentialsRequest* object. This also generates a set
+    of secrets in a **manifests** directory that is required
+    by the **aws-load-balancer-operator**.
+
+    **Note**: To Extract and prepare the `ccoctl` binary, documentation can be
+    found [here](https://docs.openshift.com/container-platform/4.10/authentication/managing_cloud_provider_credentials/cco-mode-sts.html#cco-ccoctl-configuring_cco-mode-sts).
+
+3. Apply the secrets to your cluster:
+
+    ```bash
+    ls manifests/*-credentials.yaml | xargs -I{} oc apply -f {}
```

Comment:
Are there multiple secrets or just one?

Reply:
Just one, that is just the suffix it is generated with.